docs/DeveloperGuide.md

```diff
@@ -1,14 +1,157 @@
+
+![employee class diagram v2.PNG](employee%20class%20diagram%20v2.PNG)
+
+Each `Employee` object has an employee name, and an arrayList of schedules, together with the functionality to set/get the employee name, get all of the employee's schedules, add or drop a schedule from the employee object.
+
+The arraylist of schedules contains all of the employee's schedules. Each schedule is a String which follows certain constraints as described below:
+
+A schedule must be in the format: dd/mm/yyyy
+
+dd must be an integer between 1-31
+
+mm must be an integer between 1-12
+
+yyyy must be an integer between 2021-2099
+
+The validation for a schedule is done internally in the `Employee` Class through a method called `isScheduleValid`, which is used in the `addSchedule` method in the `employee` class. Only when `isScheduleValid` returns true, the `addSchedule` method will add the schedule to the employee.
+
+The logic for allowing the user to add a new schedule is in the `EmployeeController` class in the `employee` package and the `Employee` class in the `employee` package. The sequence diagram for the interaction is shown below:
+
+![add schedule sequence diagram v2.PNG](add%20schedule%20sequence%20diagram%20v2.PNG)
+
+When the user enters “add schedule” in `AppController`, `AppController` calls the `addSchedule` method in `EmployeeController`. 
+
+In the `addSchedule` method in `EmployeeController`, the user is first asked to enter the employee’s name that the user wants to add a schedule to. 
+
+The method will then check if there is a matching employee from the arraylist of employees in the system (from the `employees` variable in `AppController`). 
+
+If no employee with the queried name is found, the method terminates with a feedback to the user. 
+
+If there is a matching employee, the system proceeds to ask the user to enter a schedule to be added for the employee. 
+
+After the user enters a schedule, control is passed to the `Employee` object, where the schedule is added to the employee if the schedule is verified to be valid. 
+
+Control is then passed back to the `addSchedule` method in `EmployeeController`, with feedback from the `Employee` object on whether the schedule is valid or not. 
+
+`addSchedule` in `EmployeeController` will provide feedback to the user on if the schedule is added successfully, and end the method call by passing control back to the `run` method in `AppController`.
+
+### Assign Employee feature
+#### Implementation
+
+The proposed Assign Employee feature is facilitated by the `Shift` object in the `shift` package,
+where it stores the employees on the shift, its shift date, shift index, and vacancy. 
+This method, like its name suggests, assigns an employee to an available shift.
+
+Given below is an example usage scenario and how the Assign Employee method behaves at each step.
+
+Step 1: The user launches the application and the `Employee` and `Shift` objects are initialised with the saved data.
```

Comment:
This step 1 is a bit hard to understand. Maybe for example this sentence 

"The example will use a `Shift` object with attributes `employees: `[Adam], 
 an ArrayList of employees assigned to this shift, meaning `Adam` is assigned to this shift, `shiftDate: `04/04/2021, `shiftIndex: `3, meaning that the shift period is 0800 to 1200, 
 and `vacancy: `2."

can be phrased as 

"The example will use a `Shift` object,  an ArrayList of employees assigned to this shift, with attributes `employees: `[Adam]. This means `Adam` is assigned to this shift, `shiftDate: `04/04/2021, `shiftIndex: `3  and `vacancy: `2. The shift index 3 indicates that the shift period is 0800 to 1200"